src/engine/controls/loopingcontrol.cpp

```diff
@@ -1212,53 +1221,48 @@ void LoopingControl::slotBeatLoop(double beats, bool keepStartPoint, bool enable
         beats = minBeatSize;
     }
 
-    int samples = static_cast<int>(m_pTrackSamples->get());
+    const auto trackEndPosition =
+            mixxx::audio::FramePos::fromEngineSamplePosMaybeInvalid(
+                    m_pTrackSamples->get());
     const mixxx::BeatsPointer pBeats = m_pBeats;
-    if (samples == 0 || !pBeats) {
+    if (!trackEndPosition.isValid() || !pBeats) {
         clearActiveBeatLoop();
         m_pCOBeatLoopSize->setAndConfirm(beats);
         return;
     }
 
-    // Calculate the new loop start and end samples
+    // Calculate the new loop start and end positios
```

Comment:
Typo. I wonder why codespell didn't detect this obvious error? Does it still work as expected?